packages/actions-shared/src/engage/utils/MessageSendPerformer.ts

```diff
@@ -523,3 +521,21 @@ export abstract class MessageSendPerformer<
     return body.traits
   }
 }
+
+type Ok<T> = { value: T }
+type Err<E> = { error: E }
+type ValueOrError<T> = Ok<T> | Err<T>
```

Comment:
looks unnecessary to add so much types, i'd just keep it as it was
```suggestion
type ValueOrError<T> = { value?: T; error?: any } 
```